### PR TITLE
Fixed browser Back Button anomaly on PDP

### LINKED ISSUE #2660: Back button behaves strangely on PDP
### Expected behavior
Back button should take you to the previous "page"

### Actual Behavior
If you have clicked on options, will take you through the history of options you clicked on

### Steps to Reproduce the Behavior
1. Go to "Basic Reaction Product"
1. Click on Red and add to cart
1. Click on Green and add to cart
1. Click on Back Button in browser
1. Notice Red is now selected rather than being taken to the Product Grid.

### Versions
```

Node: 7.10.0
NPM: 5.0.3
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: marketplace
Docker: 17.06.0-ce
```

### PR DESCRIPTION
**Pull request by aanchirinah**
Resolves #2660 ,

So there was a `Router.go` snippet in `varaintList.js` that was changing the URL when a variant is selected, so based on @mikemurray comment on how to fix the issue , I refactored the composer object (in `productDetail.js`) to get variantId by using `ReactionProduct.selectedVariant()._id` rather than fetching it as a param from the URL.

#### HOW TO TEST
- Go to "Basic Reaction Product"
- Click on Red and add to cart
- Click on Green and add to cart
- Click on Back Button of your browser and you will notice it actually goes back as it should do.

### CODE REVIEW

File: imports/plugins/included/product-detail-simple/client/containers/productDetail.js [hunk 1]
```diff
@@ -272,7 +272,7 @@ function composer(props, onData) {
   const tagSub = Meteor.subscribe("Tags");
   const shopIdOrSlug = Reaction.Router.getParam("shopSlug");
   const productId = Reaction.Router.getParam("handle");
-  const variantId = Reaction.Router.getParam("variantId");
+  const variantId = ReactionProduct.selectedVariant()._id;
```
Comment (mikemurray) [inline imports/plugins/included/product-detail-simple/client/containers/productDetail.js]: You could probably use `ReactonProduct.selectedVariantId()` instead.

Comment (aanchirinah) [reply imports/plugins/included/product-detail-simple/client/containers/productDetail.js]: Thanks I just refactored it to `selectedVariantId()`

File: imports/plugins/core/ui-navbar/client/containers/navbar.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ function composer(props, onData) {
   }
 
   if (shop && Array.isArray(shop.brandAssets)) {
-    const brandAsset = _.find(this.getShop().brandAssets, (asset) => asset.type === "navbarBrandImage");
```
Comment (zenweasel) [inline imports/plugins/core/ui-navbar/client/containers/navbar.js]: Is there a reason this change was added to this PR? If this is fixing an another issue it should be explicitly called out in the comments. Generally we should keep bugfix PR's pretty focused on fixing just the one issue.

Comment (aanchirinah) [reply imports/plugins/core/ui-navbar/client/containers/navbar.js]: I got that change in `brandAssets` when I pulled recent changes from marketplace branch.

Comment (zenweasel) [reply imports/plugins/core/ui-navbar/client/containers/navbar.js]: how would that be possible? If you are up to date with marketplace then why would you have that change? Did you merge a conflict?

Comment (aanchirinah) [reply imports/plugins/core/ui-navbar/client/containers/navbar.js]: I made a typographical error in my commit message so I reset my HEAD and unknowingly unstaged the change that was made to the `brandAssets`.

File: imports/plugins/core/ui-navbar/client/containers/navbar.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ function composer(props, onData) {
...
+    const brandAsset = _.find(shop.brandAssets, (asset) => asset.type === "navbarBrandImage");
```
Comment (spencern) [inline imports/plugins/core/ui-navbar/client/containers/navbar.js]: This is identical to the line in the current `marketplace` branch so I'm confused as to why it's showing this as a change, but it appears to be correct.
https://github.com/reactioncommerce/reaction/blob/marketplace/imports/plugins/core/ui-navbar/client/containers/navbar.js#L22

Comment (aanchirinah) [reply imports/plugins/core/ui-navbar/client/containers/navbar.js]: I pulled the latest changes from marketplace , I made a typographical error in my commit message so I reset my HEAD and unknowingly unstaged the change that was made to the `brandAssets`.

**Pull request merged**